### DISCUSSION TITLE
No log_named_bool() ?

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [foundry-rs/foundry#2436] [Help] [answered]
How can I log a bool value?

**Comment by mds1** [comment, marked as answer]
```solidity
// The first one might work, but solidity might ask you to be explicit on the types
// in which case the second one should work
emit log_named_uint("name", bool ? 1 : 0)
emit log_named_uint("name", bool ? uint256(1) : 0)
```

Alternatively, you can PR additional `log_bool` and `log_named_bool` events to forge-std: https://github.com/foundry-rs/forge-std/blob/27e14b7f2448e5f5ac32719f51fe652aa0b0733e/src/Test.sol#L20-L25

**Reply by CaledoniaProject** [reply]
I've added log_named_bool here, 
![screenshot 2022-07-23 at 07 04 02](https://user-images.githubusercontent.com/1357701/180578929-432be37e-cd93-4804-9c70-468c0a063b8a.jpg)

But it does not print anything in my test contract:

```
// SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.13;

import "forge-std/Test.sol";

contract ContractTest is Test {
    function setUp() public {}

    function testExample() public {
        emit log_named_bool("OK 123", false);
        emit log_named_address("test", address(this));
        
        assertTrue(true);
    }
}
```

Output

```
%> forge test -vvv
[⠔] Compiling...
[⠢] Compiling 1 files with 0.8.15
[⠆] Solc 0.8.15 finished in 618.42ms
Compiler run successful

Running 1 test for test/Contract.t.sol:ContractTest
[PASS] testExample() (gas: 3972)
Logs:
  test: 0xb4c79dab8f259c7aee6e5b2aa729821864227e84

Test result: ok. 1 passed; 0 failed; finished in 506.69µs
```

**Reply by mds1** [reply]
Ah right, you'll also need to open a PR to add that to the DSTest ABI in foundry. You can see another PR that made a similar change here: https://github.com/foundry-rs/foundry/pull/1916

**Reply by CaledoniaProject** [reply]
@mds1 I mean, the added `log_named_bool` event does not work for me, I don't know what's wrong

**Reply by mds1** [reply]
Right, my above comment explains what's wrong. To clarify: You can see from PR #1916 that we needed to add new bindings for the DSTest ABI—this is _in addition to_ adding the event to your local `DSTest` or `Test` contract. You need both for the event to show up in the "Logs" section.

I'll be out of town for the next week or so, so likely will not be able to get a PR in, but if you are unable to PR that fix in yourself you can always use a workaround like:

```solidity
function emitLogNamedBool(string name, bool val) internal {
  emit log_name_string(name, val ? "true" : "false");
}
```
